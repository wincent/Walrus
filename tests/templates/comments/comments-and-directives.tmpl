This is a test ## very similar to the "comments-and-text.tmpl" test
in which comments will be intermingled with directives...

## Basically, if a comment begins in the first column it produces no output at all.
## It doesn't even produce an empty line.
## The end-of-line marker after the whole-line comment is discarded.

Here's an example. This line should be immediately followed by another line.
## (because the comment on this line occupies the entire line)
Here is the other line.

But in this example, there should be a line containing only a space after this one.
 ## this comment will be thrown out but the rest of the line will survive...
Note the line above which contains only a space.

We can do the same kind of demonstration with directives...
(and placeholders, but that will be a task for another test file)

Basically if a directive produces no output...
any comment that follows it...
will be treated as if it occupied the whole line...
That means the trailing end-of-line will get thrown out as well...

Example: the next line contains a directive that produces no output...
#set foo = true ## my comment is, "this line sets foo to true"
That line should have been gobbled up in the final output.