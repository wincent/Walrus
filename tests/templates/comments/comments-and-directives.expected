This is a test 
in which comments will be intermingled with directives...


Here's an example. This line should be immediately followed by another line.
Here is the other line.

But in this example, there should be a line containing only a space after this one.
 
Note the line above which contains only a space.

We can do the same kind of demonstration with directives...
(and placeholders, but that will be a task for another test file)

Basically if a directive produces no output...
any comment that follows it...
will be treated as if it occupied the whole line...
That means the trailing end-of-line will get thrown out as well...

Example: the next line contains a directive that produces no output...
That line should have been gobbled up in the final output.